src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1449,6 +1449,46 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   simp
   exact Nat.lt_of_le_of_ne
 
+/-! ### intMin -/
+
+/-- The bitvector of width `w` that has the smallest value when interpreted as an integer. -/
+def intMin (w : Nat) : BitVec w := BitVec.ofNat w (2^(w - 1))
+
+theorem getLsb_intMin' (w : Nat) : (intMin w).getLsb i = decide (i + 1 = w) := by
+  simp only [getLsb, intMin, toNat_ofNat, Nat.testBit_mod_two_pow]
+  by_cases h : i + 1 = w
+  · subst h
+    simp
+  · by_cases i + 1 < w
+    · simp [h, @Nat.testBit_two_pow_of_ne (w-1) i (by omega)]
+    · simp only [h, decide_False]
+      simp only [Bool.and_eq_false_imp, decide_eq_true_eq] ;
+      omega
+
+theorem getLsb_intMin (w : Nat) : (intMin w).getLsb i = decide (i + 1 = w) := by
+  simp only [getLsb, intMin, toNat_ofNat, Nat.testBit_mod_two_pow]
+  rcases Nat.lt_trichotomy (i + 1) w with lt | rfl | gt
+  case inl =>
+    simp [lt, @Nat.testBit_two_pow_of_ne (w-1) i (by omega)]
+    omega
+  case inr.inl =>
+    simp
+  case inr.inr =>
+    simp [show ¬ (i + 1 = w) by omega]
+    omega
+
+@[simp, bv_toNat]
+theorem toNat_intMin : (intMin w).toNat = 2^(w - 1) % 2^w := by
+  simp [intMin]
+
+@[simp]
+theorem neg_intMin {w : Nat} : -intMin w = intMin w := by
+  by_cases h : 0 < w
+  · simp [bv_toNat, h]
+  · simp only [Nat.not_lt, Nat.le_zero_eq] at h
+    subst h
+    simp [intMin]
```

Comment:
How is "subst" different from "rw" or "simp"? (I tried to look up the documentation, did not find much). If "subst" just provides substitution, then maybe this proof could be shortened to 
```lean
simp [h, intMin]
```

Reply:
`subst` does indeed "just provide substitution". However in certain DTT hell scenarios `subst` is often able to perform substitutions along an equality while `simp` and `rw` cannot due to the way that they work. Whether that is the case here I do not know.

Reply:
I tried, but here the order seems to be relevant.